Autores: ....

### EDA

#### Normalización de datos y validación

* Clasificación de región en Regiones y Ciudades

* Explorar datos, fechas, nulos, columnas innecesarias, (outlayers)

* Discusión sobre qué hacer con los datos de semanas que faltan (interpolante)

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import avocado_manager as av
av.init(); #display(av.df("df_type"))
df_cp = av.df("df_cp")
df_cp_cleaned = df_cp
display(df_cp)

index       Date  AveragePrice  Total Volume  Volume_Hass_S  \
0          0 2015-12-27          1.33      64236.62        1036.74   
1          1 2015-12-20          1.35      54876.98         674.28   
2          2 2015-12-13          0.93     118220.22         794.70   
3          3 2015-12-06          1.08      78992.15        1132.00   
4          4 2015-11-29          1.28      51039.60         941.48   
...      ...        ...           ...           ...            ...   
18244  18244 2018-02-04          1.63      17074.83        2046.96   
18245  18245 2018-01-28          1.71      13888.04        1191.70   
18246  18246 2018-01-21          1.87      13766.76        1191.92   
18247  18247 2018-01-14          1.93      16205.22        1527.63   
18248  18248 2018-01-07          1.62      17489.58        2894.77   

       Volume_Hass_L  Volume_Hass_XL  Total Bags  Small Bags  Large Bags  \
0           54454.85           48.16     8696.87     8603.62       93.25   
1           44638.81           58.33     9505.56     9408.07       97.49   
2          109149.67          130.50     8145.35     8042.21      103.14   
3           71976.41           72.58     5811.16     5677.40      133.76   
4           43838.39           75.78     6183.95     5986.26      197.69   
...              ...             ...         ...         ...         ...   
18244        1529.20            0.00    13498.67    13066.82      431.85   
18245        3431.50            0.00     9264.84     8940.04      324.80   
18246        2452.79          727.94     9394.11     9351.80       42.31   
18247        2981.04          727.01    10969.54    10919.54       50.00   
18248        2356.13          224.53    12014.15    11988.14       26.01   

       XLarge Bags          type  year            region region_class  
0              0.0  conventional  2015            Albany         City  
1              0.0  conventional  2015            Albany         City  
2              0.0  conventional  2015            Albany         City  
3              0.0  conventional  2015            Albany         City  
4              0.0  conventional  2015            Albany         City  
...            ...           ...   ...               ...          ...  
18244          0.0       organic  2018  WestTexNewMexico       Region  
18245          0.0       organic  2018  WestTexNewMexico       Region  
18246          0.0       organic  2018  WestTexNewMexico       Region  
18247          0.0       organic  2018  WestTexNewMexico       Region  
18248          0.0       organic  2018  WestTexNewMexico       Region  

[18249 rows x 15 columns]

In [7]:
# Numero de registros(Count) de Total Volume agrupnado por tipo.
display(av.df("df_type"))

# Utiliza info() para revisar la estructura y los tipos de datos del conjunto.
df_cp = av.df("df_cp")
print(df_cp.info())

# Realiza un análisis exploratorio para entender la estructura del conjunto, incluyendo 
# el número de filas y columnas, tipos de datos y valores faltantes. 
# Imprime la cantidad de valores faltantes por columna utilizando isnull().

#Forma del DataFrame (filas, columnas)
data_shape = df_cp.shape
print(f"Forma del DataFrame: {data_shape}")

column_names = df_cp.columns
print(f"Nombres de Columnas: {column_names}")

# Tipos de datos de cada columna
data_types = df_cp.dtypes
print(f"Tipos de Datos:\n{data_types}")

#df.isnull
#missing_values = pd.isnull(df)
print(f"Cantidad de nulls: {df_cp.isnull().sum()}")

# series de tiempo
df_cp['Date'] = pd.to_datetime(df_cp['Date'])
ventas_mensual = df_cp.groupby(df_cp['Date'].dt.to_period("M"))['Total Volume'].sum()
display(ventas_mensual)

# Resumen Estadístico
data_summary = df_cp.describe()
print("\nResumen Estadístico:")
print(data_summary)

df_cp.head(5)

type  Total Volume
0  conventional          9126
1       organic          9123

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18249 entries, 0 to 18248
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   index           18249 non-null  int64         
 1   Date            18249 non-null  datetime64[ns]
 2   AveragePrice    18249 non-null  float64       
 3   Total Volume    18249 non-null  float64       
 4   Volume_Hass_S   18249 non-null  float64       
 5   Volume_Hass_L   18249 non-null  float64       
 6   Volume_Hass_XL  18249 non-null  float64       
 7   Total Bags      18249 non-null  float64       
 8   Small Bags      18249 non-null  float64       
 9   Large Bags      18249 non-null  float64       
 10  XLarge Bags     18249 non-null  float64       
 11  type            18249 non-null  object        
 12  year            18249 non-null  int64         
 13  region          18249 non-null  object        
 14  region_class    18249 non-null  object        
dtypes:

Date
2015-01    3.180852e+08
2015-02    3.626376e+08
2015-03    4.185831e+08
2015-04    3.441444e+08
2015-05    4.837840e+08
2015-06    3.818267e+08
2015-07    3.567019e+08
2015-08    4.192448e+08
2015-09    3.292188e+08
2015-10    3.057761e+08
2015-11    3.694400e+08
2015-12    2.960259e+08
2016-01    4.754058e+08
2016-02    4.286458e+08
2016-03    3.912124e+08
2016-04    4.018328e+08
2016-05    5.612305e+08
2016-06    4.103282e+08
2016-07    4.672753e+08
2016-08    3.707970e+08
2016-09    3.623033e+08
2016-10    3.602018e+08
2016-11    2.602093e+08
2016-12    3.314478e+08
2017-01    5.333330e+08
2017-02    4.714672e+08
2017-03    3.589034e+08
2017-04    4.898145e+08
2017-05    4.257305e+08
2017-06    4.104107e+08
2017-07    4.749037e+08
2017-08    3.383330e+08
2017-09    2.847795e+08
2017-10    3.627780e+08
2017-11    3.245193e+08
2017-12    4.593329e+08
2018-01    4.297071e+08
2018-02    4.977782e+08
2018-03    4.552531e+08
Freq: M, Name: Total Volume, dtype: float64


Resumen Estadístico:
              index                           Date  AveragePrice  \
count  18249.000000                          18249  18249.000000   
mean    9124.000000  2016-08-13 23:30:43.498273792      1.405978   
min        0.000000            2015-01-04 00:00:00      0.440000   
25%     4562.000000            2015-10-25 00:00:00      1.100000   
50%     9124.000000            2016-08-14 00:00:00      1.370000   
75%    13686.000000            2017-06-04 00:00:00      1.660000   
max    18248.000000            2018-03-25 00:00:00      3.250000   
std     5268.176867                            NaN      0.402677   

       Total Volume  Volume_Hass_S  Volume_Hass_L  Volume_Hass_XL  \
count  1.824900e+04   1.824900e+04   1.824900e+04    1.824900e+04   
mean   8.506440e+05   2.930084e+05   2.951546e+05    2.283974e+04   
min    8.456000e+01   0.000000e+00   0.000000e+00    0.000000e+00   
25%    1.083858e+04   8.540700e+02   3.008780e+03    0.000000e+00   
50%    1.073768e+05 

index       Date  AveragePrice  Total Volume  Volume_Hass_S  Volume_Hass_L  \
0      0 2015-12-27          1.33      64236.62        1036.74       54454.85   
1      1 2015-12-20          1.35      54876.98         674.28       44638.81   
2      2 2015-12-13          0.93     118220.22         794.70      109149.67   
3      3 2015-12-06          1.08      78992.15        1132.00       71976.41   
4      4 2015-11-29          1.28      51039.60         941.48       43838.39   

   Volume_Hass_XL  Total Bags  Small Bags  Large Bags  XLarge Bags  \
0           48.16     8696.87     8603.62       93.25          0.0   
1           58.33     9505.56     9408.07       97.49          0.0   
2          130.50     8145.35     8042.21      103.14          0.0   
3           72.58     5811.16     5677.40      133.76          0.0   
4           75.78     6183.95     5986.26      197.69          0.0   

           type  year  region region_class  
0  conventional  2015  Albany         City  
1  conventional  2015  Albany         City  
2  conventional  2015  Albany         City  
3  conventional  2015  Albany         City  
4  conventional  2015  Albany         City

In [ ]:
display(pd.unique(av.df("df_cp")['region']))
display(av.df("df_city").groupby(["region"])["Total Volume"].sum().reset_index().sort_values("Total Volume").tail())

display(av.df("df_region").groupby(["region"])["Total Volume"].sum().reset_index().sort_values("Total Volume").tail())
display(av.df("df_city_region").groupby(["region"])["Total Volume"].sum().reset_index().sort_values("Total Volume").tail())
display(av.df("df_greater").groupby(["region"])["Total Volume"].sum().reset_index().sort_values("Total Volume").tail())


#df_sort = av.sort("df_cp_greater","region"); display(df_sort)
#display(av.df("df_cp_totalUS")) #.unique("region"))
df_largest = av.df("region_largest")
display(df_largest)

#df = av.df("df_cp");df.info()
#display(av.df("df_type"))




array(['Albany', 'Atlanta', 'BaltimoreWashington', 'Boise', 'Boston',
       'BuffaloRochester', 'California', 'Charlotte', 'Chicago',
       'CincinnatiDayton', 'Columbus', 'DallasFtWorth', 'Denver',
       'Detroit', 'GrandRapids', 'GreatLakes', 'HarrisburgScranton',
       'HartfordSpringfield', 'Houston', 'Indianapolis', 'Jacksonville',
       'LasVegas', 'LosAngeles', 'Louisville', 'MiamiFtLauderdale',
       'Midsouth', 'Nashville', 'NewOrleansMobile', 'NewYork',
       'Northeast', 'NorthernNewEngland', 'Orlando', 'Philadelphia',
       'PhoenixTucson', 'Pittsburgh', 'Plains', 'Portland',
       'RaleighGreensboro', 'RichmondNorfolk', 'Roanoke', 'Sacramento',
       'SanDiego', 'SanFrancisco', 'Seattle', 'SouthCarolina',
       'SouthCentral', 'Southeast', 'Spokane', 'StLouis', 'Syracuse',
       'Tampa', 'TotalUS', 'West', 'WestTexNewMexico'], dtype=object)

region  Total Volume
25  SanFrancisco  1.358302e+08
7         Denver  1.389025e+08
10       Houston  2.031679e+08
17       NewYork  2.407341e+08
14    LosAngeles  5.078965e+08

region  Total Volume
6     MiamiFtLauderdale  9.767322e+07
0   BaltimoreWashington  1.347139e+08
13     WestTexNewMexico  1.445218e+08
9         PhoenixTucson  1.956433e+08
3         DallasFtWorth  2.084193e+08

region  Total Volume
29  PhoenixTucson  1.956433e+08
16        Houston  2.031679e+08
10  DallasFtWorth  2.084193e+08
25        NewYork  2.407341e+08
20     LosAngeles  5.078965e+08

region  Total Volume
6     Southeast  6.152384e+08
3     Northeast  7.132809e+08
5  SouthCentral  1.011280e+09
0    California  1.028982e+09
7          West  1.086779e+09

#### Visión global de datos

Analisis de Series Temporales y Ruido asociado
Precios promedio calibre 
Precios promedio por bolsa
Separar alvocats convencionals i alvocats orgànics
Comparación precios promedio convencional y organico
Representación de ventas totales sobre precio promedio ( por regiones o no )

### ANALISIS

#### Separar alvocats convencionals i alvocats orgànics

#### Impacte del preu en les vendes

#### Correlacion de les "grans regions" i regions amb volum total US

#### Correlacion de volum total de les ciutats amb volum total US

#### Elasticidad

#### Analisis por Cohortes

#### Matriz de correlaciones de todas las variables según  apartado 5.6. En especial, precio medio y relación con diferentes calibres.

#### Dispersión entre variables claves según 5.2

### REGRESIONES Y PROYECCIONES

#### Creación del modelo de regresión

#### Creación de modelo predictivo de ventas por MES usando los datos de años anteriores ( lineal y polinomica ) Calculo R2

#### Creación de modelo predictivo de ventas por TRIMESTRE usando los datos de años anteriores ( lineal y polinomica ) Calculo R2

#### Creación de modelo predictivo de ventas por AÑO usando los datos de años anteriores ( lineal y polinomica ) Calculo R2

#### Modelos de regresión multiple entre todas las variables según  apartado 5.6. En especial, precio medio y relación con diferentes calibres.Precio vs Total Bags

#### Analisis de coeficientes de regresion multiple. Afectación de cada variable al AvgPrice apartado 5.7

#### Modelos de regresión para diferenciar volumenes de ventas Apartado 5.8 . Calibres y AveragePrice.

#### Creación de modelo de regresión y polinómica de ventas totales sobre precio promedio ( por regiones o no ) apartado 5.9

#### Predicciones

#### Prediccion de precio promedio según volumen aguacates y por tipo ( y por calibres ??) . Lineal y Polinomica. Calculo de R2